appliances/MinIO/appliance.sh

```diff
@@ -0,0 +1,396 @@
+    'ONEAPP_MINIO_TLS_CERT'            'configure'  'MinIO TLS certificate (.crt)'                                  'O|text64'
+    'ONEAPP_MINIO_TLS_KEY'             'configure'  'MinIO TLS key (.key)'                                          'O|text64'
+)
+
+
+### Appliance metadata ###############################################
+
+# Appliance metadata
+ONE_SERVICE_NAME='Service MinIO - KVM'
+ONE_SERVICE_VERSION=''   #latest
+ONE_SERVICE_BUILD=$(date +%s)
+ONE_SERVICE_SHORT_DESCRIPTION='Appliance with preinstalled MinIO for KVM hosts'
+ONE_SERVICE_DESCRIPTION=$(cat <<EOF
+Appliance with preinstalled MinIO.
+
+This appliance installs the latest version of MinIO DEB from the official download mirror. If no parameters are defined
+in the template contextualization, the MinIO Environment Variable File will use the defaults defined [here]
+(https://min.io/docs/minio/linux/operations/install-deploy-manage/deploy-minio-single-node-single-drive.html#create-the-environment-variable-file).
+
+This appliance can deploy a Single-Node Single-Drive or a Single-Node Multi-Drive installation of MinIO. The configuration script
+will automatically detect how many drives are present on the VM and partition, format and mount all detected disks.
+If just one disk is detected, the appliance will configure MinIO as Single-Node Multi-Drive. If more than one additional disk are detected,
+the appliance will configure as Single-Node Multi-Drive.
+
+Notes for "Single-Node Multi-Drive":
+  - At least 4 drives have to be configured on the VM template, as that it is the minimum number of drives indicated in the [documentation]
+  (https://min.io/docs/minio/linux/operations/checklists/hardware.html#id2).
+  - All drives have to be the same size since MinIO limits the size used per drive to the smallest drive in the deployment.
+  Refer to MinIO documentation for additional information.
+  - The appliance will use /dev/vdb as the first disk for MinIO storage, and continue with vdc, vdd, etc. If any disk has a XFS partition already created,
+it asumes the configuration was already made in previous configuration steps. It will skip the partition and formatting of the drive.
+
+WARNING: After initial configuration do not add / remove disks to the VM since the configuration scripts will not
+manage storage configuration changes.
+EOF
+)
+ONE_SERVICE_RECONFIGURABLE=true
+
+# ------------------------------------------------------------------------------
+# Contextualization defaults
+#  - "AG", is for "AutoGenerated" certs
+# ------------------------------------------------------------------------------
+
+ONEAPP_MINIO_ROOT_USER="${ONEAPP_MINIO_ROOT_USER:-myminioadmin}"
+ONEAPP_MINIO_ROOT_PASSWORD="${ONEAPP_MINIO_ROOT_PASSWORD:-minio-secret-key-change-me}"
+ONEAPP_MINIO_OPTS="${ONEAPP_MINIO_OPTS:---console-address :9001}"
+MINIO_HOSTNAME="${ONEAPP_MINIO_HOSTNAME:-localhost,minio-*.example.net}"
+MINIO_TLS_CERT="${ONEAPP_MINIO_TLS_CERT}"
```

Comment:
Is there a reason for introducing additional variables MINIO_TLS_CERT + KEY?  I think they aren't used by minio itself, right? :thinking: